### ISSUE TITLE
Progress event is never received on queue

### ISSUE THREAD

**Issue by JakubPetriska** [taskforcesh/bullmq#94]
Progress event is never received on queue object. In the example below `'Progress'` will never be printed.
```javascript
myQueue.on(
  'progress',
  () => {
    console.log('Progress')
  }
)
```

I'm initializing the event handler right after creating the queue object. Is there something wrong with my code?

BullMQ version: 1.6.2

**Comment by stansv** [comment]
In BullMQ progress event is fired by worker, not Queue:

```
const test = async () => {
  const worker = new Worker("test", async job => {
    let count = 10;
    while (count-- > 0) {
      await job.updateProgress(10 - count);
    }
  });

  worker.on('progress', (job, progress) => {
      console.log('Progress' + JSON.stringify(progress));
    }
  );

  await new Queue("test").add("any", {});
};
```

**Comment by JakubPetriska** [comment]
Documentation contains this sample https://docs.bullmq.io/guide/events :
```javascript
import { Queue } from 'bullmq'

const myQueue = new Queue('Paint');

myQueue.on('waiting', (job: Job) => {
  // Job is waiting to be processed.
});
```
So that should probably be removed.

**Comment by manast** [comment]
@JakubPetriska the ```waiting``` event is indeed emitted by the Queue class :). But ```progress``` is emitted by the worker. If you want you can also use global events to listen to all the events from the QueueEvents class...

**Comment by JakubPetriska** [comment]
@manast What if you need to listen to progress events on certain queue from all workers?

**Comment by stansv** [comment]
@JakubPetriska this should have been answer to your question 😎—

> you can also use global events to listen to all the events from the QueueEvents class...

**Comment by JakubPetriska** [comment]
I get that but than I would need to prefix job ids with queue names and check that right? If not how would you be sure that the event comes from certain queue?

**Comment by stansv** [comment]
@JakubPetriska no, each queue has its own event stream. When you create QueueEvents object you must provide queue name in constructor.

**Comment by JakubPetriska** [comment]
Ok thanks for clarification. I guess it would make sense to add to documentation which event is emitted by what if it's not already there.

**Comment by JakubPetriska** [comment]
Hi @manast and @stansv I figured out a few suggestions for the events page to clear up the confusion that I went through with the progress event and queue events in general. If you guys want I can do the PR for these changes. Otherwise here are the suggestions that would IMHO resolve these issues:

- Add `QueueEvents` object construction to the `QueueEvents` usage example. This helps because due to the missing constructor call I thought that `QueueEvents` is some kind of static object.
- Fix parameters of the `progress` event handler. This confused me later on as the handler is accepting different arguments than what is mentioned in the documentation.
